example/project/cluster/integration/bootstrap_test.lua

```diff
@@ -0,0 +1,217 @@
+                alias = 'tnt-storage-1-replica',
+                env = {
+                    ['TARANTOOL_APP_NAME'] = 'example-app',
+                    ['TARANTOOL_LOG_LEVEL'] = 6,
+                }
+            },
+        }
+    },
+    {
+        uuid = cluster_helpers.uuid('c'),
+        roles = { 'vshard-storage', 'app.roles.custom' },
+        servers = {
+            {
+                instance_uuid = cluster_helpers.uuid('c', 1),
+                alias = 'tnt-storage-2-master',
+                env = {
+                    ['TARANTOOL_APP_NAME'] = 'example-app',
+                    ['TARANTOOL_LOG_LEVEL'] = 6,
+                }
+            },
+            {
+                instance_uuid = cluster_helpers.uuid('c', 2),
+                alias = 'tnt-storage-2-replica',
+                env = {
+                    ['TARANTOOL_APP_NAME'] = 'example-app',
+                    ['TARANTOOL_LOG_LEVEL'] = 6,
+                }
+            }
+        },
+    }
+}
+
+local instances = {}
+for _, replicaset in ipairs(replicasets) do
+    for _, server in ipairs(replicaset.servers) do
+        table.insert(instances, server.alias)
+    end
+end
+
+test.before_suite(function()
+    group.cluster = cluster_helpers.Cluster:new({
+        datadir = helper.datadir,
+        server_command = helper.server_command,
+        use_vshard = true,
+        replicasets = replicasets
+    })
+
+    group.cluster:start()
+    group.cluster:upload_config({
+        metrics = {
+            export = {
+                {
+                    path = '/metrics',
+                    format = 'json'
+                },
+                {
+                    path = '/metrics/prometheus',
+                    format = 'prometheus'
+                }
+            }
+        }
+    })
+end)
+
+test.after_suite(function()
+    group.cluster:stop()
+end)
+
+local function http_request(alias, method, endpoint, count)
+    if count <= 0 then
+        return
+    end
+
+    for _ = 1, count do
+        group.cluster:server(alias):http_request(method, endpoint, { raise = false })
+    end
+end
+
+local last_key = {
+    ['tnt-router'] = 1,
+    ['tnt-storage-1-master'] = 1,
+    ['tnt-storage-2-master'] = 1,
+}
+
+local charset = {} -- [0-9a-zA-Z]
+for c = 48, 57  do table.insert(charset, string.char(c)) end
+for c = 65, 90  do table.insert(charset, string.char(c)) end
+for c = 97, 122 do table.insert(charset, string.char(c)) end
+
+local function random_string(length)
+    if not length or length <= 0 then return '' end
+    math.randomseed(os.clock()^5)
+    return random_string(length - 1) .. charset[math.random(1, #charset)]
+end
+
+local function space_operations(alias, operation, count)
+    if count <= 0 then
+        return
+    end
+
+    local space = group.cluster:server(alias).net_box.space.MY_SPACE
+
+    if operation == 'select' then
+        for _ = 1, count do
+            space[operation](space, {}, { limit = 1 })
```

Comment:
You could replace this with actual space method calls, e.g. space:select(....) This could make code much simpler to read.